plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,910 @@ type CiscoTelemetryMDT struct {
+		str = strconv.FormatUint(val.Uint64Value, 10)
+		return str
+	case *telemetry.TelemetryField_Sint32Value:
+		str = strconv.FormatInt(int64(val.Sint32Value), 10)
+		return str
+	case *telemetry.TelemetryField_Sint64Value:
+		str = strconv.FormatInt(val.Sint64Value, 10)
+		return str
+	}
+	return ""
+}
+
+//xform Uint64 to int64
+func nxosValueXformUint64Toint64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	if field.GetUint64Value() != 0 {
+		return int64(value.(uint64))
+	}
+	return nil
+}
+
+//xform string to float
+func nxosValueXformStringTofloat(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//convert property to float from string.
+	vals := field.GetStringValue()
+	if vals != "" {
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+	}
+	return nil
+}
+
+//xform string to uint64
+func nxosValueXformStringToUint64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//string to uint64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//xform string to int64
+func nxosValueXformStringToInt64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//string to int64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//auto-xform
+func nxosValueAutoXform(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	} // switch
+	return nil
+}
+
+//auto-xform float properties
+func nxosValueAutoXformFloatProp(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+	} // switch
+	return nil
+}
+
+//xform uint64 to string
+func nxosValueXformUint64ToString(field *telemetry.TelemetryField, value interface{}) interface{} {
+	switch val := field.ValueByType.(type) {
+	case *telemetry.TelemetryField_StringValue:
+		if len(val.StringValue) > 0 {
+			return val.StringValue
+		}
+	case *telemetry.TelemetryField_Uint64Value:
+		return (strconv.FormatUint(val.Uint64Value, 10))
+	}
+	return nil
+}
+
+//Xform value field.
+func (c *CiscoTelemetryMDT) nxosValueXform(field *telemetry.TelemetryField, value interface{}, path string) interface{} {
+	if strings.ContainsRune(path, ':') {
+		// not NXOS
+		return nil
+	}
+	if _, ok := c.propMap[field.Name]; ok {
+		return (c.propMap[field.Name](field, value))
```

Comment:
Please remove the extra parentheses
```suggestion
		return c.propMap[field.Name](field, value)
```